### ISSUE TITLE
Add spec for BatchNormInferenceOp

### ISSUE THREAD

**Issue by sdasgup3** [openxla/stablehlo#347]
### Request description

To spec for stablehlo.batch_norm_inference
